translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5410,8 +5410,20 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
   }
 
   case OpConstantNull: {
-    auto LT = transType(BV->getType());
-    return mapValue(BV, Constant::getNullValue(LT));
+    auto BVTy = BV->getType();
+    auto LT = transType(BVTy);
+    Value *NullVal;
+    // For local memory space (LDS) the NULL value is 0xFFFFFFFF, not 0x0.
+    if (BVTy->isTypePointer() && BVTy->getPointerStorageClass() == spv::StorageClassWorkgroup)
```

Comment:
Follow existing if-else style
if () {
  ...
} else
 ...

This style will be changed when we rework the translator.